Server/src/services/authenticationService.ts

```diff
@@ -119,12 +120,38 @@ export class AuthenticationService {
         return token;
     }
 
+    async validateUserDetails(userDetailUpdater: IUserDetailUpdater): Promise<IResponse> {
+        let email = userDetailUpdater.email
+        let password = userDetailUpdater.password
+        let organization = userDetailUpdater.organization
+
+        // email = await AuthenticationModel.verifyIfEmailExists(userDetailUpdater.email);
+        if (await AuthenticationModel.verifyIfEmailExists(email)) {
+            try {
+                if (password != undefined) {
+                    password = await this.hashPassword(password)
+                    await AuthenticationModel.updateUserPasswordDetail(email, password);
+                }
+                if (organization != undefined) {
+                    await AuthenticationModel.updateUserOrganizationDetail(email, organization);
+                }
+            } catch (error) {
+                throw new InternalServerError("Internal server problem...try again!");
+            }
+        }
+        else {
+            throw new BadRequest("Email is not valid. Please enter the correct email!");
```

Comment:
We can make this message a bit better since by now the user has already logged in. The only time we say email is not valid - please enter correct is when they are trying to login.

Something like: Email cant be found, request declined - use error 406 maybe

Reply:
i updated the message and the test is now running. Thank you for the feedback ;)